rest_framework/permissions.py

```diff
@@ -172,7 +172,7 @@ def get_required_object_permissions(self, method, model_cls):
         return [perm % kwargs for perm in self.perms_map[method]]
 
     def has_object_permission(self, request, view, obj):
-        model_cls = view.queryset.model
+        model_cls = view.get_queryset().model
```

Comment:
Should we also fallback to `view.queryset.model`? That'd be consistent with the documentation above, and would be less likely to introduce any breakage to existing APIs.